msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -145,53 +145,16 @@
  * </p>
  * </p>
  */
-public final class PublicClientApplication {
+public class PublicClientApplication implements IPublicClientApplication, IMultipleAccountPublicClientApplication {
```

Comment:
Long-term I think we would benefit from splitting this into 4 classes - this way implementations are thematically grouped and we minimize the risk of developers internally calling the wrong functions. We also then prohibit the developer from casting the result and doing weird things

- 1 Factory class to produce subclasses
    * Factory should have an `interface` and a default implementation
- 1 abstract superclass with common methods for the various PCA subclasses
- 1 interface + 1 subclass for single-user
- 1 interface + 1 subclass for multi-user

Reply:
Based on the design doc, we were originally planning to split them into 3 classes.

- IMultipleAccountPublicClientApplication will be splitted into another class in the next CR. (Keep it here temporarily so that the test app doesn't break).
- ISingleAccountPublicClientApplication will also be another class.
- PublicClientApplication & the Factory is still in the same class

I agree we could/should split them into separate cards. Could do that in the upcoming PRs.